### PR TITLE
fix custom `assetFileNames` issue

### PR DESCRIPTION
**Pull request by apatel369**
## Changes
closes[ #426](https://github.com/withastro/adapters/issues/426) in adapters repo

When using custom `assetFileNames` in Astro configuration assets were incorrectly moved to the server directory instead of the client directory. This caused 404 errors when the browser tried to load these assets from the client side.

Users could specify custom directories for various asset types (e.g., CSS, images). To handle that used asset paths from the build output to ensure assets are moved from/to the correct directories.

## Testing

Added tests for testing image and CSS assets with custom paths. 
The previous test was passing cause accidentally `build.assets` and custom `assetFileNames` path were matching.
([`assets`](https://github.com/withastro/astro/blob/6031962ab5f56457de986eb82bd24807e926ba1b/packages/astro/test/fixtures/custom-assets-name/astro.config.mjs#L27) and [assets/css](https://github.com/withastro/astro/blob/6031962ab5f56457de986eb82bd24807e926ba1b/packages/astro/test/fixtures/custom-assets-name/astro.config.mjs#L20C54-L20C66))
So updated that as well. 

## Docs

N/A bug fix

### CODE REVIEW

File: packages/astro/src/core/build/static-build.ts [hunk 1]
```diff
@@ -470,30 +475,36 @@ export async function copyFiles(fromFolder: URL, toFolder: URL, includeDotfiles
...
 		opts.settings.config.output === 'static'
 			? opts.settings.config.build.client
 			: opts.settings.config.build.server;
 	const clientRoot = opts.settings.config.build.client;
 	const assets = opts.settings.config.build.assets;
-	const serverAssets = new URL(`./${assets}/`, appendForwardSlash(serverRoot.toString()));
-	const clientAssets = new URL(`./${assets}/`, appendForwardSlash(clientRoot.toString()));
+	let serverAssets = new URL(`./${assets}/`, appendForwardSlash(serverRoot.toString()));
+	let clientAssets = new URL(`./${assets}/`, appendForwardSlash(clientRoot.toString()));
+	if (opts?.settings?.config?.vite?.build?.rollupOptions?.output?.assetFileNames) {
```
Comment (apatel369) [inline packages/astro/src/core/build/static-build.ts:488]: Not sure this is the best way to check user has defined custom assetFileNames. I am open to suggestions for a better approach.

Comment (apatel369) [inline packages/astro/src/core/build/static-build.ts:488]: I'm not sure this is the best way to check if the user has defined custom assetFileNames. I am open to suggestions for a better approach.

File: .changeset/neat-papayas-brake.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+Fixes an issue where custom `assetFileNames` were not working as expected.
```
Comment (ematipico) [inline .changeset/neat-papayas-brake.md:5]: Can you provide a better description, please? "not working as expected" is a bit broad. You provided a good description in the PR, and maybe some of that description could be moved here. 

For example, we should tell the users in which case the `assetFileNames` were not working (when read from the client?).

**Review by ematipico — APPROVED** [review]
Thank you @apatel369, sorry for the late review. The fix looks good, I would suggest a better description for merging the PR

File: .changeset/neat-papayas-brake.md (hunk 2 above)
Comment (apatel369) [reply .changeset/neat-papayas-brake.md]: Thanks for the feedback. I updated it.

**Pull request merged**